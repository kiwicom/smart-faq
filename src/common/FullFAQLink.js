import React from "react";
import { Typography } from "@kiwicom/orbit-components";
import { OpenInNew } from "@kiwicom/orbit-components/lib/icons";
import css from "styled-jsx/css";

const style = css`
  div.open-icon {
    display: inline-block;
    vertical-align: -3px;
    margin-left: 4px;
  }

  a {
    text-decoration: none;
    color: inherit;
  }

  .primary {
    color: #00a991;
  }
`

const FullFAQLink = ({ className }) => (
  <div>
    <Typography type="attention">
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://www.kiwi.com/helpcenter/"
        className={className}
      >
        Full FAQ site
        <span className="inline-icon">
          <div className="open-icon">
            <OpenInNew fill={className === 'primary' ? '#00a991' : '#171b1e'} height="16" />
          </div>
        </span>
      </a>
    </Typography>
    <style jsx>{style}</style>
  </div>
);

export default FullFAQLink;
