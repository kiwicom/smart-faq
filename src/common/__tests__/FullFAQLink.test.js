// @flow
import * as React from 'react';
import renderer from 'react-test-renderer';

import FullFAQLink from '../FullFAQLink';

describe('FullFAQLink', () => {
  it('should match snapshot', () => {
    const result = renderer.create(<FullFAQLink />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
