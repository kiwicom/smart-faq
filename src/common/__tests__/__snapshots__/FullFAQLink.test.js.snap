// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullFAQLink should match snapshot 1`] = `
<div
  className="jsx-760019255"
>
  <span
    className="jsx-410270121"
  >
    <a
      className="jsx-760019255 "
      href="https://www.kiwi.com/helpcenter/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Full FAQ site
      <span
        className="jsx-760019255 inline-icon"
      >
        <div
          className="jsx-760019255 open-icon"
        >
          <svg
            fill="#171b1e"
            height="16"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.667 4v2.667h4.786L10.347 19.773l1.88 1.88L25.333 8.547v4.786H28V4h-9.333zm6.666 21.333H6.667V6.667H16V4H6.667C5.187 4 4 5.2 4 6.667v18.666A2.667 2.667 0 0 0 6.667 28h18.666A2.667 2.667 0 0 0 28 25.333V16h-2.667v9.333z"
            />
          </svg>
        </div>
      </span>
    </a>
  </span>
</div>
`;
